ui/src/app/views/breadcrumb/component.ts

```diff
@@ -65,43 +70,60 @@ export class BreadcrumbComponent implements OnInit, OnDestroy {
   private buildBreadcrumbs(route: ActivatedRoute,
                            url: string = '',
                            breadcrumbs: Breadcrumb[] = []): Array<Breadcrumb> {
-    const ROUTE_DATA_BREADCRUMB = 'breadcrumb';
     const children: ActivatedRoute[] = route.children;
     if (children.length === 0) {
       return breadcrumbs;
     }
     for (const child of children) {
-      if (!child.snapshot.data.hasOwnProperty(ROUTE_DATA_BREADCRUMB)) {
+      if ((!child.snapshot.data.hasOwnProperty(this.ROUTE_DATA_BREADCRUMB))
+          && (!child.snapshot.data.hasOwnProperty(this.ROUTE_DATA_INTERMIEDIATE_BREADCRUMB))) {
         return this.buildBreadcrumbs(child, url, breadcrumbs);
       }
       const routeURL: string = child.snapshot.url.map(segment => segment.path).join('/');
       if (routeURL.length > 0) {
         url += `/${routeURL}`;
       }
-      let label = child.snapshot.data[ROUTE_DATA_BREADCRUMB];
+
+      let label;
+      let breadcrumbType;
+      if (child.snapshot.data[this.ROUTE_DATA_BREADCRUMB] != null) {
+        label = child.snapshot.data[this.ROUTE_DATA_BREADCRUMB];
+        breadcrumbType = this.ROUTE_DATA_BREADCRUMB;
+      } else {
+        label = child.snapshot.data[this.ROUTE_DATA_INTERMIEDIATE_BREADCRUMB];
+        breadcrumbType = this.ROUTE_DATA_INTERMIEDIATE_BREADCRUMB;
+      }
+
       if (label === 'Param: Workspace Name') {
         label = child.snapshot.data['workspace'].name;
       }
       if (label === 'Param: Cohort Name') {
         label = child.snapshot.data['cohort'].name;
       }
-      if (label === 'Param: Concept Sets Name') {
-        label = child.snapshot.data['workspace'].name + ' Concepts';
-        if (breadcrumbs.length > 2) {
-          breadcrumbs = breadcrumbs.filter(b => !b.url.endsWith('/concepts'));
-        }
-      }
       if (label === 'Param: Concept Set Name') {
         label = child.snapshot.data['conceptSet'].name;
       }
       // Prevent processing children with duplicate urls
       if (!breadcrumbs.some(b => b.url === url)) {
-        const breadcrumb = BreadcrumbComponent.makeBreadcrumb(label, url, child);
+        const breadcrumb = BreadcrumbComponent.makeBreadcrumb(label, breadcrumbType, url, child);
         breadcrumbs.push(breadcrumb);
       }
       return this.buildBreadcrumbs(child, url, breadcrumbs);
     }
+  }
 
+  /**
+   * Filters an array of Breadcrumbs so that the last element is never an intermediateBreadcrumb
+   * This ensures that breadcrumb headers are displayed correctly while still tracking
+   * intermediate pages.
+   */
+  private filterBreadcrumbs(breadcrumbs: Breadcrumb[]): Array<Breadcrumb> {
+    let last = breadcrumbs[breadcrumbs.length - 1];
```

Comment:
nit: should probably handle the 0 length breadcrumbs edge case

Reply:
the empty list?